src/components/header.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect, useRef } from 'react';
+
+interface HeaderProps {
+  newTodo: string;
+  loading: boolean;
+  onAddTodo: (event: React.FormEvent) => void;
```

Comment:
Same as for update todo - this function should accept new todo title as an argument (state for it you can have here - in Header component)